agent/agent.go

```diff
@@ -4668,7 +4669,12 @@ func (a *Agent) persistServerMetadata() {
 				continue
 			}
 
-			f.Close()
+			sf := f.(*safeio.File)
+			if err := sf.Commit(); err != nil {
```

Comment:
nit:

it might be better to use the guarded version, so that if for some reason your `io.WriteCloser` was different it would still work:
```
if sf, ok := f.(*safeio.File); ok {
   if err := sf.Commit(); err != nil {
      // ... stuff...
     }
}
```

Reply:
Updated. Thanks.